node/pkg/aggregator/aggregator.go

```diff
@@ -0,0 +1,228 @@
+package aggregator
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"strconv"
+	"sync"
+	"time"
+
+	"bisonai.com/orakl/node/pkg/raft"
+	"bisonai.com/orakl/node/pkg/utils"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/libp2p/go-libp2p/core/host"
+)
+
+const RoundSync raft.MessageType = "roundSync"
+const PriceData raft.MessageType = "priceData"
+
+type Aggregator struct {
+	Raft  *raft.Raft
+	Host  host.Host
+	Ps    *pubsub.PubSub
+	Topic *pubsub.Topic
+	Sub   *pubsub.Subscription
+
+	LeaderJobTicker *time.Ticker
+	JobTicker       *time.Ticker
+
+	LeaderJobTimeout *time.Duration
+	JobTimeout       *time.Duration
+
+	CollectedPrices map[int][]int
+	AggregatorMutex sync.Mutex
+
+	RoundID int
+}
+
+type RoundSyncMessage struct {
+	LeaderID string `json:"leaderID"`
+	RoundID  int    `json:"roundID"`
+}
+
+type PriceDataMessage struct {
+	RoundID   int `json:"roundID"`
+	PriceData int `json:"priceData"`
+}
+
+func NewAggregator(h host.Host, ps *pubsub.PubSub, topicString string) (*Aggregator, error) {
+	topic, err := ps.Join(topicString)
+	if err != nil {
+		return nil, err
+	}
+
+	sub, err := topic.Subscribe()
+	if err != nil {
+		return nil, err
+	}
+
+	leaderTimeout := 5 * time.Second
+
+	_aggregator := Aggregator{
```

Comment:
I am curious. Does the prefix underscore character have some meaning?

Reply:
awh, the code used to be different, first declaring it on _aggregator and then reinitializing to aggregator, since the code is updated, I'll remove it